### ISSUE TITLE
idle and hard timeout for flows don't work

### ISSUE THREAD

**Issue by franciscodelicado** [MeshSr/onetswitch30#1]
Hi,
I am trying to program flows with an idle_timeout and hard_timeout in a OneSwitch30 board. But it doesn't work well:
- The idle_timeout expires although the flow is used all time, so the OFPFlowRemoved message is generated even though the flow receives traffic. But the flow is not really removed from flow-table, because the messages which match with this flow are forwarding like before the OFPFlowRemoved message generation.
  - The hard_timeout expires and a OFPFlowRemove message is generated correctly. But as in idle_timeout case, the flow entry is not removed from flow-table.

Could you help me with any idea?

Best regards.

**Comment by franciscodelicado** [comment]
Sorry!! I forgot include my system information:

zynq> uname -a
Linux (none) 3.12.0-xilinx-g61f47a8-dirty #1 SMP PREEMPT Fri Apr 3 15:32:02 CST 2015 armv7l GNU/Linux

zynq> ./udatapath/ofdatapath --version
add default entries to table 0,prio:0
add default entries to table 0,prio:1
add default entries to table 0,prio:2
add default entries to table 0,prio:3
add default entries to table 0,prio:4
add default entries to table 0,prio:5
add default entries to table 0,prio:6
add default entries to table 0,prio:7
ofdatapath 1.3.0 compiled Mar 25 2015 11:00:58

zynq> ./secchan/ofprotocol --version
ofprotocol 1.3.0 compiled Mar 25 2015 11:00:54

zynq> ./utilities/dpctl --version
dpctl 1.3.0 compiled Mar 25 2015 11:00:54

Regards

**Comment by myazzurro** [comment]
Dear Francisco,

Guess you were using the ofs-hw project, i.e. ons30-app52-ref_ofshw, right?
In this case, by now, our hardware-implemented flow table cannot support timeout scheme:(
ONetSwitch users need to modify the ARM software(https://github.com/MeshSr/ofs-hw), by translating the OFPFlowRemoved message to the register read/write via AMBA to FPGA logic.

Thanks.
Best regards,

**Comment by franciscodelicado** [comment]
OK. Thanks.
Only one more question,

It is possible delete a flow using a OPFPFlowMod message? 
In this case, my controller could generate periodically events and delete manually the flows. It will be a manual solution, but could work for me.

Thanks myazzurro, for your help.

Regards

**Comment by myazzurro** [comment]
Yes, deleting a flow is supported. You can try it using _dpctl_ as the example below

```
./utilities/dpctl tcp:127.0.0.1:6632 flow-mod cmd=dels,table=0,prio=15 in_port=2
#or
./utilities/dpctl tcp:127.0.0.1:6632 flow-mod cmd=del,table=0
```
